pkg/providers/vsphere/reconciler/reconciler_test.go

```diff
@@ -334,6 +336,33 @@ func TestReconcilerReconcileControlPlaneFailure(t *testing.T) {
 	tt.Expect(err).To(HaveOccurred())
 }
 
+func TestReconciler_ReconcileStorageCLass(t *testing.T) {
+	tt := newReconcilerTest(t)
+	tt.createAllObjs()
+
+	logger := test.NewNullLogger()
+	remoteClient := env.Client()
+	spec := tt.buildSpec()
+
+	tt.remoteClientRegistry.EXPECT().GetClient(
+		tt.ctx, client.ObjectKey{Name: "workload-cluster", Namespace: "eksa-system"},
+	).Return(remoteClient, nil)
+
+	result, err := tt.reconciler().InstallStorageCLass(tt.ctx, logger, spec)
+
+	tt.Expect(err).NotTo(HaveOccurred())
+	tt.Expect(tt.cluster.Status.FailureMessage).To(BeZero())
+	tt.Expect(result).To(Equal(controller.Result{}))
+
+	namespacedName := types.NamespacedName{
+		Name:      "standard",
+		Namespace: "eksa-system",
+	}
+	obj := &storagev1.StorageClass{}
+	err = remoteClient.Get(tt.ctx, namespacedName, obj)
```

Comment:
Is this an applicable scenario to use `tt.ShouldEventuallyExist` in the test instead of calling `Get` here?

Reply:
`client.get` does the same thing, but I can use `ShouldEventuallyExist` for consistency